site/build/reference/associate.js

```diff
@@ -78,17 +79,22 @@ function associateCast(property, creation, methods) {
   }
 }
 
-function associateProperty(property, creation, methods) {
+function associateProperty(property, creation, methods, references) {
   if (property.directObjectType) {
     assertNotTuple(property.directObjectType);
     methods[property.directObjectType].properties.push(property.key);
+    references[property.key] = property.directObjectType;
   } else if (property.indexType) {
-    unique(types(property.indexType)).forEach(function(type) {
```

Comment:
In my opinion, this is better expressed as:

```javascript
var indexTypes = unique(types(property.indexType));
references[property.key] = indexTypes[0];
indexTypes.forEach(...);
```

At any rate, no spaces inside parens.